execution/distinct.go

```diff
@@ -24,104 +28,162 @@ func (node *Distinct) Get(ctx context.Context, variables octosql.Variables, stre
 		return nil, nil, errors.Wrap(err, "couldn't get source stream ID")
 	}
 
-	stream, execOutput, err := node.child.Get(ctx, variables, sourceStreamID)
+	source, execOutput, err := node.source.Get(ctx, variables, sourceStreamID)
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "couldn't get stream for child node in distinct")
+		return nil, nil, errors.Wrap(err, "couldn't get stream for source node in distinct")
+	}
+
+	trigger, err := NewCountingTrigger(NewDummyValue(octosql.MakeInt(1))).Get(ctx, variables)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "couldn't create trigger for distinct")
+	}
+
+	distinct := &DistinctStream{
+		variables: variables,
+	}
+
+	processFunc := &ProcessByKey{
+		eventTimeField:  "", // TODO: what about this?
```

Comment:
You have to handle it similarly as in group by. Using the event time for distinct is important, as it allows distinct usage on streams.